lib/Doctrine/ruleset.xml

```diff
@@ -30,4 +30,147 @@
             <property name="ignoreNewlines" value="true"/>
         </properties>
     </rule>
+
+    <!-- Forbid using absolute class name references (except global ones) -->
+    <rule ref="vendor/slevomat/coding-standard/SlevomatCodingStandard/Sniffs/Namespaces/ReferenceUsedNamesOnlySniff.php">
+        <properties>
+            <property name="allowFullyQualifiedGlobalClasses" type="boolean" value="true"/>
+            <property name="allowFullyQualifiedNameForCollidingClasses" type="boolean" value="true"/>
+        </properties>
+    </rule>
+
+    <!-- Forbid empty lines around type declarations -->
+    <rule ref="vendor/slevomat/coding-standard/SlevomatCodingStandard/Sniffs/Types/EmptyLinesAroundTypeBracesSniff.php">
+        <properties>
+            <property name="linesCountAfterOpeningBrace" value="0"/>
+            <property name="linesCountBeforeClosingBrace" value="0"/>
+        </properties>
+    </rule>
+
+    <!-- Require space around colon in return types -->
+    <rule ref="vendor/slevomat/coding-standard/SlevomatCodingStandard/Sniffs/TypeHints/ReturnTypeHintSpacingSniff.php">
+        <properties>
+            <property name="spacesCountBeforeColon" value="1"/>
+        </properties>
+    </rule>
+
+    <!-- Require types to be written as natively if possible;
+         require iterable types to specify phpDoc with their contnet;
+         forbid useless/duplicated information in php doc -->
+    <rule ref="vendor/slevomat/coding-standard/SlevomatCodingStandard/Sniffs/TypeHints/TypeHintDeclarationSniff.php">
```

Comment:
This sniff is pretty advanced:
* Requires native types be expressed natively (i.e. scalar types).
* Requires iterable types (`array`/`iterable`/custom) have specified what they contain, i.e. `@return Foo[]`.
* Forbids duplication of type information - all `@param` and `@return` are forbidden unless they're mixed, traversable or explicit description is given.

Reply:
That last one looks awesome